explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+
+![Validating Resolver](../images/ubuntu-local-validating-resolver.png)
+
+This is the case if you install the `bind9` DNS server: the default configuration is to act as a validating resolver. This can be seen in `/etc/bind/named.conf.options` after installing the `bind9` package:
+
+    options {
+        ...
+        dnssec-validation auto;
+        ...
+    };
+
+The key aspect of this deployment type is that the network between the stub resolver and the validating resolver has to be trusted, otherwise this is not useful. The Validating Resolver will be performing all the DNSSEC checks, and only returning a response if it was validated, but that response could have been altered on that "last mile" network segment.
+
+This is where the `trusted-ad` setting from `/etc/resolv.conf` comes into play:
+
+    nameserver 127.0.0.53
+    options edns0 trust-ad
+
+The `trust-ad` setting is documented in the [resolv.conf(5)](https://manpages.ubuntu.com/manpages/noble/man5/resolv.conf.5.html) manpage. It means that the local resolver will:
+
+ * Set the `ad` bit (Authenticated Data) in the outgoing queries.
+ * Trust the `ad` bit in the responses from the specified nameserver.
+
+When the `ad` bit is set in a DNS response, it means that DNSSEC validation was performed and successful. The data was authenticated.
+
+Specifyhing `trust-ad` in `/etc/resolv.conf` implies in these assumptions:
+
+ * The 127.0.0.53 name server is trusted to set the `ad` flag correctly in its responses. If it performs DNSSEC validation, it is trusted to perform this validation correctly, and set the `ad` flag accordingly. If it does not perform DNSSEC validation, then the `ad` flag will always be unset in the responses.
+ * The network path between localhost and 127.0.0.53 is trusted.
+
+When using systemd-resolved as the stub resolver, like in the config above, the network path is the localhost interface, and that is inherently trusted. But the actual nameserver that will be used is not 127.0.0.53: it will be whatever systemd-resolved was configured with. Unless local DNSSEC validation is enabled, systemd-resolved will remove the `ad` bit in the queries it forwards to the validating resolver, and also remove it from the responses.
+
+This is the default case in Ubuntu systems.
+
+Another valid configuration is to not use systemd-resolved, but point at the validating resolver of the network directly, like in this example:
+
+    nameserver 10.10.17.11
+    options edns0 trust-ad
+
+The `trust-ad` config means exactly the same as before: the `ad` bit will be set in outgoing queries, and its value in the responses will be trusted. But here the name server is no longer localhost, but another IP on the network. These are the assumptions being made by this config, same as before:
```

Comment:
"The `trust-ad` configuration functions similarly to the previous scenario. The 'ad' bit is included in outgoing queries, and the resolver trusts the `ad` bit in incoming responses. However, in this case, the nameserver is located at a different IP address on the network. This configuration relies on the same assumptions as before:"

Reply:
adopted slightly modified